### DISCUSSION TITLE
Equations not rendering properly

### DISCUSSION THREAD

**Discussion by rsbyrne** [executablebooks/meta#362] [Q&A] [answered]
Me *yet again* - this time about equations.

My equations are rendering just fine in the usual Jupyter notebook markdown cell context:
![image](https://user-images.githubusercontent.com/11673079/118384288-4404d500-b648-11eb-9389-d987c4e19e35.png)
(screengrab of https://github.com/rsbyrne/thesis/blob/main/book/content/chapter_02_methods/section1.ipynb)

But are all borked on the actual Github pages upload:
![image](https://user-images.githubusercontent.com/11673079/118384272-2b94ba80-b648-11eb-96d9-454deae12e0c.png)
(screengrab of https://rsbyrne.github.io/thesis/content/chapter_02_methods/section1.html)

Presumably the LaTeX engine that Jupyter Book is using has lesser or different syntax support for equations. Some of them render okay, but many don't. I'm sure there's a way to configure this, but I don't know how.

My versions:
![image](https://user-images.githubusercontent.com/11673079/118384390-02c0f500-b649-11eb-8d91-61cf1b62b53b.png)

Thanks for all the help - y'all are really saving my bacon.

-Rohan

**Comment by choldgraf** [comment]
If you use dollar signs does it render? It looks like you don't have the [ ] set to render as math in jupyter book. You may need to manually set this in sphinx. See here for how this is done in myst: https://myst-parser.readthedocs.io/en/latest/using/syntax.html#mathjax-and-math-parsing

**Reply by rsbyrne** [reply]
Hmm, that can't be it because whereas this renders fine:
`\\[ Pr \equiv \frac{\nu_r}{\kappa_r} = \frac{\frac{\mu_r}{\rho_r}}{\frac{k_r}{\rho_r c_{p_r}}} \\]`

This does not:
`\\[ Nu = 1 + \left| \frac{\partial \theta^*}{\partial y^*} \right| _S \\]`

Scroll down the page and you'll find heaps of examples: some work, some don't, and I can't figure out what the pattern is:
https://rsbyrne.github.io/thesis/content/chapter_02_methods/section1.html

My config file is here FYI:
https://github.com/rsbyrne/thesis/blob/main/book/_config.yml

I'll keep poking around.

**Reply by rsbyrne** [reply]
Okay, so it turns out that actually *none* of the equations are rendering anymore: the seemingly ad hoc quality of the problem was just due to caching of notebook cells.

So I must have some general Sphinx problem on my hands.

On a related note, is there a Docker container or anything like that that would demonstrate a properly configured environment for a Jupyter Book? Presumably the official Jupyter Book documentation is built by some VM somewhere - if I knew how that machine is configured, I could compare it to my own build and figure out what I'm missing.

**Reply by rsbyrne** [reply]
Alright, so I seem to have them working now. The key was forcing an upgrade to `myst-parser`. It raised a bunch of scary error messages saying that Jupyter Book would prefer the older version, but the newer version caused everything to render properly immediately.

My myst-parser version is now:
![image](https://user-images.githubusercontent.com/11673079/118385698-9b10a700-b654-11eb-8547-71069e97f1a5.png)

Is it possible that Jupyter Book's requirements are out of date?

**Reply by choldgraf** [reply]
@rsbyrne definitely possible, we are about to release a new version of Jupyter Book and I think some other versions will upgrade soon that will result in more alignment

**Reply by rsbyrne** [reply]
Pumped for the next release. Thanks so much to you and the other maintainers for driving forward this transformative project.

**Comment by rsbyrne** [comment, marked as answer]
TL;DR, the key was forcing an upgrade to `myst-parser`:
`pip3 install -U myst-parser`

In general, while doing so seems kind of slapdash, I'm having a lot of success with just forcing upgrades to everything as soon as I run into a problem!
